inc/Cache/Cache.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace dokuwiki\Cache;
+
+/**
+ * Generic handling of caching
+ */
+class Cache
+{
+    public $key = '';          // primary identifier for this item
+    public $ext = '';          // file ext for cache data, secondary identifier for this item
+    public $cache = '';        // cache file name
+    public $depends = array(); // array containing cache dependency information,
+    //   used by _useCache to determine cache validity
+
+    public $_event = '';       // event to be triggered during useCache
+    public $_time;
+    public $_nocache = false;  // if set to true, cache will not be used or stored
+
+    /**
+     * @param string $key primary identifier
+     * @param string $ext file extension
+     */
+    public function __construct($key, $ext)
+    {
+        $this->key = $key;
+        $this->ext = $ext;
+        $this->cache = getCacheName($key, $ext);
+    }
+
+    /**
+     * public method to determine whether the cache can be used
+     *
+     * to assist in centralisation of event triggering and calculation of cache statistics,
+     * don't override this function override _useCache()
+     *
+     * @param  array $depends array of cache dependencies, support dependecies:
+     *                            'age'   => max age of the cache in seconds
+     *                            'files' => cache must be younger than mtime of each file
+     *                                       (nb. dependency passes if file doesn't exist)
+     *
+     * @return bool    true if cache can be used, false otherwise
+     */
+    public function useCache($depends = array())
+    {
+        $this->depends = $depends;
+        $this->addDependencies();
+
+        if ($this->_event) {
+            return $this->stats(trigger_event($this->_event, $this, array($this, 'makdeDefaultCacheDecision')));
+        } else {
+            return $this->stats($this->makdeDefaultCacheDecision());
+        }
+    }
+
+    /**
+     * private method containing cache use decision logic
+     *
+     * this function processes the following keys in the depends array
+     *   purge - force a purge on any non empty value
+     *   age   - expire cache if older than age (seconds)
+     *   files - expire cache if any file in this array was updated more recently than the cache
+     *
+     * Note that this function needs to be public as it is used as callback for the event handler
+     *
+     * can be overridden
+     *
+     * @internal This method may only be called by the event handler! Call \dokuwiki\Cache\Cache::useCache instead!
+     *
+     * @return bool               see useCache()
+     */
+    public function makdeDefaultCacheDecision()
```

Comment:
- This function was named `_useCache()` before. I like and understand renaming it to a self-explaining name. 
- However, the first part of the name is a bit strange for me. Is it the verb 'make' with a typo? or is this a German verb? Please improve.
- Please rename all the occurrences of the old `_useCache`.

Reply:
There is a disconnect with the comment and the method visbility too.  One should really be changed to match the other.

Reply:
@Klap-in I really didn't see that extra "d" until you pointed it out, thanks :)
I also renamed that method in two comments that still referred to `_useCache`. Did you see the old comment name anywhere else?